src/Processors/Transforms/FilterTransform.cpp

```diff
@@ -137,7 +153,11 @@ void FilterTransform::doTransform(Chunk & chunk)
     constant_filter_description = ConstantFilterDescription(*filter_column);
 
     if (constant_filter_description.always_false)
+    {
```

Comment:
I think the next condition (`if (constant_filter_description.always_true)`) is redundant. This is not the result of your PR, but with your changes it became confusing, because you are rightfully not handling the profile events there. Could you please remove that?

Reply:
Removed the `if (constant_filter_description.always_true)` block altogether.